src/accelerate/utils/memory.py

```diff
@@ -174,7 +174,17 @@ def get_xpu_available_memory(device_index: int):
             from intel_extension_for_pytorch.xpu import mem_get_info
 
             return mem_get_info(device_index)[0]
+    elif version.parse(torch.__version__).release >= version.parse("2.6").release:
+        # torch.xpu.mem_get_info API is available starting from PyTorch 2.6
+        # It further requires PyTorch built with the SYCL runtime which supports API
+        # to query available device memory. If not available, exception will be
+        # raised.
```

Comment:
I've updated comment a little to clarify the story around SYCL version. It's tight to the version of DPC++ SYCL compiler used to build pytorch with XPU backend support. It's reported with `print(torch.version.xpu)` and the first version to support required feature is `20250001`. I think link on how to build is not necessary here in accelerate comments since that's a generic link to pytorch documentation on how to build it with XPU backend support. I.e. https://github.com/pytorch/pytorch?tab=readme-ov-file#intel-gpu-support. It further links to intel side documentation to install required environment which includes DPC++ SYCL compiler (this one: https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html).

Note that I deliberately used `try...catch` here instead of version checks around `torch.version.xpu` since I think that's better approach. SYCL support is getting upstream to clang and in the future there might be at least 2 compilers supporting it - intel dpc++ one and clang. I think using `try...catch` would handle such situation in accelerate counting for the future. We can discuss this further if needed.